app/src/main/java/io/appium/uiautomator2/model/internal/CustomUiDevice.java

```diff
@@ -94,7 +94,23 @@ public UiAutomation getUiAutomation() {
         return getInstrumentation().getUiAutomation();
     }
 
-    private UiObject2 toUiObject2(Object selector, AccessibilityNodeInfo node) {
+    private UiObject2 toUiObject2(@Nullable Object selector, @Nullable AccessibilityNodeInfo node) {
+        if (selector == null) {
+            // FIXME: The 'selector' should be proper By instance as non-null in the interaction.
+            Logger.debug("FIXME: selector argument should not be null in androidx.test.uiautomator:uiautomator:2.3.0");
+        }
+        if (node == null && android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.R) {
```

Comment:
wouldn't it be safer to always assign `node` to an empty object if it is not , so we could get rid of this condition?

Reply:
The api level comparison? Then the constructor is available since api level 30.

Reply:
> Then the constructor is available since api level 30.

then we probably have no other choice if the constructor is not there in older API version.
it would then make sense to change the code and introduce a new variable, so we don't reassign the method argument. See https://sapegin.me/blog/avoid-reassigning-variables/

Reply:
Sounds good, changed to define a new variable